bitsandbytes/research/autograd/_functions.py

```diff
@@ -186,7 +184,9 @@ def backward(ctx, grad_output):
 
 class SwitchBackBnb(torch.autograd.Function):
     @staticmethod
-    def forward(ctx, A, B, out=None, bias=None, state=MatmulLtState()):
+    # TODO: the B008 on the line below is a likely bug; the current implementation will
+    #       have each SwitchBackBnb instance share a single MatmulLtState instance!!!
```

Comment:
Thanks for raising this. We should open an issue for this. I'll put it on my list, but feel free to open one if you feel like it.